pkg/oc/admin/diagnostics/config.go

```diff
@@ -35,3 +24,117 @@ func (o DiagnosticsOptions) buildRawConfig() (*clientcmdapi.Config, error) {
 	}
 	return &kubeConfig, nil
 }
+
+// determine if we even have a client config
+func (o DiagnosticsOptions) detectClientConfig() (expected bool, detected bool) {
+	if o.ClientFlags == nil {
+		// options for client not provided, so it must not be expected.
+		return false, false
+	}
+	o.Logger().Notice("CED2011", "Determining if client configuration exists for client/cluster diagnostics")
+	confFlagName := config.OpenShiftConfigFlagName
+	confFlagValue := o.ClientFlags.Lookup(confFlagName).Value.String()
+	successfulLoad := false
+
+	var foundPath string
+	rules := config.NewOpenShiftClientConfigLoadingRules()
+	paths := append([]string{confFlagValue}, rules.Precedence...)
+	for index, path := range paths {
+		errmsg := ""
+		switch index {
+		case 0:
+			errmsg = fmt.Sprintf("--%s specified that client config should be at %s\n", confFlagName, path)
+		case len(paths) - 1: // config in ~/.kube
+		// no error message indicated if it is not there... user didn't say it would be
+		default: // can be multiple paths from the env var in theory; all cases should go here
+			if len(os.Getenv(config.OpenShiftConfigPathEnvVar)) != 0 {
+				errmsg = fmt.Sprintf("Env var %s specified that client config could be at %s\n", config.OpenShiftConfigPathEnvVar, path)
+			}
+		}
+
+		if o.canOpenConfigFile(path, errmsg) && foundPath == "" {
+			successfulLoad = true
+			foundPath = path
+		}
+	}
+	if foundPath != "" {
+		if confFlagValue != "" && confFlagValue != foundPath {
+			// found config but not where --config said
+			o.Logger().Error("DCli1001", fmt.Sprintf(`
```

Comment:
should this be a fatal error? if a user provides an explicit flag for a config file location, but we instead infer and find it elsewhere, is it safe to assume we are using a configuration that the user expects?

I get the sense that we may be doing too much for the user in this case.
As a user, I would probably be fine with this failing out if I provide a `--config` value and it does not point to an actual config, and only have the actual configuration location discovered if I don't explicitly provide this flag

Reply:
The effect is that all diagnostics requiring the client config are skipped. The other ones can still run, and you get a bad exit code and this message about your options to get client config working. I don't think it's necessary to actually halt execution at this point though I could easily see that argument.

Reply:
Originally I was trying to think of users who don't even know where their kubeconfig is...

Reply:
> Originally I was trying to think of users who don't even know where their kubeconfig is

That is a fair point. Would it make sense to provide a list of found locations containing a kubeconfig in this case?

Reply:
It does do that (although it's a short list), but does not actually use any found kubeconfig, as that crosses the line (IMHO) into "too helpful".